{
 "cells": [
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Matrix 합치기\n",
    "- FASTQ로부터 align 한 다음 htseq count 한 raw count를 받아보자\n",
    "- Single end : Fibula, Mandible/Maxilla/Tibia, ENCODE Osteocyte\n",
    "- 우리 데이터 : \"/data/project/OPLL/0.rnaraw/Analysis/~230316/00.PCA&Demographics/df/df_coladd.txt\"\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "from Bio import SeqIO\n",
    "from biomart import BiomartServer\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "import os\n",
    "import palettable\n",
    "import rpy2.robjects as robjects\n",
    "from rpy2.robjects.packages import importr\n",
    "\n",
    "# Import R packages in Python\n",
    "base = importr('base')\n",
    "utils = importr('utils')\n",
    "deseq2 = importr('DESeq2')\n",
    "# sva = importr (\"sva\")    # for Combat_seq\n",
    "\n",
    "#from reComBat import reComBat\n",
    "\n",
    "tabl = palettable.tableau.Tableau_20.mpl_colors"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "- GSE175236 (ENCODE Osteocyte),   GSE78608 (ENCODE Osteoblast), GSE227994 (Oxford_Ob)\n",
    "- GSE220630 (Fibula, 5개), GSE149167 (maxilla, mandible, tibia)\n",
    "- GSE188760 (OPLL, CSM, 2023, 4개), GSE69787(OPLL, Lig, 2016, 4개), GSE186542 (NP, 6개),\n",
    "- GSE197172 (BM-MSC, 3개),  GSE102312 (BM-MSC_sb, 6개)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [],
   "source": [
    "batch_list = []\n",
    "batch_index = 0\n",
    "\n",
    "# Display the filenames\n",
    "for project in [\"GSE175236\", \"GSE220630\",  \"GSE149167\" , \"GSE188760\", \"GSE69787\", \"GSE186542\", \"GSE197172\"]:\n",
    "    DIR =  \"/data/project/OPLL/0.rnaraw/99.public/\" + project + \"/03.matrix\"\n",
    "    for filename in sorted(os.listdir(DIR))  :\n",
    "        if os.path.isfile (os.path.join ( DIR, filename)) == False:\n",
    "            continue\n",
    "        if (project ==\"GSE149167\") :\n",
    "            if ( filename.split(\"_\")[1].split(\".tsv\")[0] in ([\"4\", \"5\", \"6\"])  ):\n",
    "                continue\n",
    "            \n",
    "        if (\"tsv\" in filename) & (\"norm_counts\" not in filename):\n",
    "            INPUT_TSV = DIR + \"/\" + filename\n",
    "            #print (\"{}\\t{}\".format (filename, df.shape ) )\n",
    "            SAMPLE_ID = filename.split(\".\")[0]\n",
    "\n",
    "            df = pd.read_csv (INPUT_TSV, sep = \"\\t\", names = [\"GENE_SYMBOL\", SAMPLE_ID] )\n",
    "\n",
    "            batch_list.append( batch_index )\n",
    "\n",
    "            if len(batch_list) == 1:  # 맨 처음이라면\n",
    "                df_concat = df\n",
    "            else:\n",
    "                df_concat = pd.merge (df_concat, df,  how = \"outer\").fillna (0)\n",
    "\n",
    "\n",
    "    batch_index = batch_index + 1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [],
   "source": [
    "# # GSE220630 (Fibula) : single end 라서 어쩔 수 없이 그냥 matrix를 합쳐주자\n",
    "\n",
    "# import pandas as pd\n",
    "# import numpy as np\n",
    "# import pyensembl\n",
    "\n",
    "# pyensemble = pyensembl.EnsemblRelease(77)   # release 77 uses human reference genome GRCh38\n",
    "\n",
    "# INPUT_TSV = \"/data/project/OPLL/0.rnaraw/99.public/GSE220630/03.matrix/GSE220630_norm_counts.tsv\"\n",
    "# df = pd.read_csv (INPUT_TSV, sep = \"\\t\").iloc[ : , [0, 1, 6, 13, 19] ]  # 4개만 골라주자\n",
    "# df.columns = [\"ENSEMBL_ID\", \"Fibula_1\", \"Fibula_2\", \"Fibula_3\", \"Fibula_4\"]\n",
    "# df [\"GENE_SYMBOL\"] = \"\"\n",
    "\n",
    "# for i in range (df.shape[0]):\n",
    "#     E_ID = df.iloc [i, 0]\n",
    "#     try:\n",
    "#         #print ( \"{}\\t{}\".format(E_ID, pyensemble.gene_name_of_gene_id( E_ID ) ) )\n",
    "#         df.iloc[i, -1] = pyensemble.gene_name_of_gene_id( E_ID )\n",
    "#     except:\n",
    "#         #print ( \"{} : No Gene Symbol\".format (E_ID))\n",
    "#         print ( \"\", end = \"\")\n",
    "    \n",
    "#     if i % 5000 == 0:\n",
    "#         print (\"\", end = \" \")\n",
    "\n",
    "\n",
    "# df_concat = pd.merge (df_concat, df ,   left_on = \"GENE_SYMBOL\", right_on = \"GENE_SYMBOL\", how = \"left\").fillna (0)\n",
    "# # Reorder DataFrame\n",
    "# cols = df_concat.columns.tolist()\n",
    "# cols.insert(1, cols.pop(cols.index('ENSEMBL_ID')))\n",
    "# df_concat = df_concat[cols] \n",
    "# df_concat = df_concat.iloc[:-5, ]\n",
    "\n",
    "# # Update batch_list\n",
    "# batch_list = batch_list + [batch_index] * 4\n",
    "# batch_index = batch_index + 1"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "\n",
    "### GTEX의 whole blood data 4개를 합쳐주자"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [],
   "source": [
    "df = pd.read_csv ( \"/data/project/OPLL/0.rnaraw/99.public/GTEX/gene_reads_2017-06-05_v8_whole_blood.txt\",  sep = \"\\t\")\n",
    "\n",
    "NUM_SELECT_GTEX = 4\n",
    "df = df.iloc [:, [2] + list( range(3, 3 + NUM_SELECT_GTEX)) ]\n",
    "df.columns = [\"GENE_SYMBOL\"] + [\"WB_{}\".format(i) for i in range(1, df.shape[1])]\n",
    "#df.head()\n",
    "df_concat = pd.merge (df_concat, df ,   left_on = \"GENE_SYMBOL\", right_on = \"GENE_SYMBOL\", how = \"left\").fillna (0)\n",
    "\n",
    "batch_list = batch_list + [batch_index] * NUM_SELECT_GTEX\n",
    "batch_index = batch_index + 1"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Our data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Out data 개수 : 30\n",
      "batch_list : [0, 0, 0, 0, 1, 1, 1, 1, 1, 2, 2, 2, 2, 2, 2, 2, 2, 2, 3, 3, 3, 3, 4, 4, 4, 4, 4, 4, 5, 5, 5, 5, 5, 5, 6, 6, 6, 7, 7, 7, 7, 8, 9, 8, 9, 8, 9, 8, 9, 8, 9, 8, 9, 8, 9, 8, 9, 8, 9, 10, 12, 8, 9, 10, 12, 8, 9, 10, 11, 8, 9] (71개)\n"
     ]
    }
   ],
   "source": [
    "df = pd.read_csv ( \"/data/project/OPLL/0.rnaraw/Analysis/~230316/00.PCA&Demographics/df/df_coladd.txt\",  sep = \"\\t\")\n",
    "coldata = [ int(line.rstrip('\\n')) for line in open(\"/data/project/OPLL/0.rnaraw/Analysis/~230316/00.PCA&Demographics/coldata/group.txt\", 'r')]\n",
    "batch_list = batch_list +  list ( coldata + np.max(batch_list) )\n",
    "\n",
    "print (\"Out data 개수 : {}\\nbatch_list : {} ({}개)\".format (len(coldata) , batch_list, len(batch_list)))\n",
    "\n",
    "df_concat = pd.merge (df_concat, df ,   left_on = \"GENE_SYMBOL\", right_on = \"GENE_SYMBOL\", how = \"left\").fillna (0)\n",
    "\n",
    "df_concat.iloc[:, 1:] = df_concat.iloc[:, 1:].astype(int)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [],
   "source": [
    "#df_concat.drop([\"ENSEMBL_ID\"], axis = 1).to_csv (\"/data/project/OPLL/0.rnaraw/99.public/df_concat.tsv\", sep = \"\\t\", index = False, header = True)\n",
    "df_concat.to_csv (\"/data/project/OPLL/0.rnaraw/99.public/df_concat.tsv\", sep = \"\\t\", index = False, header = True)\n",
    "pd.DataFrame(batch_list).to_csv (\"/data/project/OPLL/0.rnaraw/99.public/batch_list.tsv\", sep = \"\\t\", index = False, header = False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>GENE_SYMBOL</th>\n",
       "      <th>ENCODE_Os_1</th>\n",
       "      <th>ENCODE_Os_2</th>\n",
       "      <th>ENCODE_Os_3</th>\n",
       "      <th>ENCODE_Os_4</th>\n",
       "      <th>fibula_1</th>\n",
       "      <th>fibula_2</th>\n",
       "      <th>fibula_3</th>\n",
       "      <th>fibula_4</th>\n",
       "      <th>fibula_5</th>\n",
       "      <th>...</th>\n",
       "      <th>230207-NORMAL</th>\n",
       "      <th>230207-OLF</th>\n",
       "      <th>230207-SPINOUS</th>\n",
       "      <th>230222-LF</th>\n",
       "      <th>230222-NORMAL</th>\n",
       "      <th>230222-OLF</th>\n",
       "      <th>230222-SPINOUS</th>\n",
       "      <th>230316-DISH</th>\n",
       "      <th>230316-NORMAL</th>\n",
       "      <th>230316-OPLL</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A1BG</td>\n",
       "      <td>0</td>\n",
       "      <td>102</td>\n",
       "      <td>91</td>\n",
       "      <td>90</td>\n",
       "      <td>30</td>\n",
       "      <td>21</td>\n",
       "      <td>41</td>\n",
       "      <td>17</td>\n",
       "      <td>80</td>\n",
       "      <td>...</td>\n",
       "      <td>298</td>\n",
       "      <td>812</td>\n",
       "      <td>388</td>\n",
       "      <td>173</td>\n",
       "      <td>393</td>\n",
       "      <td>771</td>\n",
       "      <td>105</td>\n",
       "      <td>1151</td>\n",
       "      <td>642</td>\n",
       "      <td>480</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>A1BG-AS1</td>\n",
       "      <td>1</td>\n",
       "      <td>41</td>\n",
       "      <td>28</td>\n",
       "      <td>50</td>\n",
       "      <td>13</td>\n",
       "      <td>3</td>\n",
       "      <td>17</td>\n",
       "      <td>3</td>\n",
       "      <td>14</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>11</td>\n",
       "      <td>0</td>\n",
       "      <td>60</td>\n",
       "      <td>54</td>\n",
       "      <td>6</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>55</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>A1CF</td>\n",
       "      <td>23</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>5</td>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "      <td>3</td>\n",
       "      <td>5</td>\n",
       "      <td>...</td>\n",
       "      <td>16</td>\n",
       "      <td>27</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>77</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>A2M</td>\n",
       "      <td>6</td>\n",
       "      <td>18</td>\n",
       "      <td>12</td>\n",
       "      <td>14</td>\n",
       "      <td>7667</td>\n",
       "      <td>7103</td>\n",
       "      <td>5452</td>\n",
       "      <td>2909</td>\n",
       "      <td>16884</td>\n",
       "      <td>...</td>\n",
       "      <td>15229</td>\n",
       "      <td>31998</td>\n",
       "      <td>20449</td>\n",
       "      <td>77759</td>\n",
       "      <td>14516</td>\n",
       "      <td>21324</td>\n",
       "      <td>6991</td>\n",
       "      <td>36400</td>\n",
       "      <td>52678</td>\n",
       "      <td>66246</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>A2M-AS1</td>\n",
       "      <td>4</td>\n",
       "      <td>18</td>\n",
       "      <td>26</td>\n",
       "      <td>17</td>\n",
       "      <td>39</td>\n",
       "      <td>43</td>\n",
       "      <td>23</td>\n",
       "      <td>16</td>\n",
       "      <td>66</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>82</td>\n",
       "      <td>10</td>\n",
       "      <td>11</td>\n",
       "      <td>0</td>\n",
       "      <td>26</td>\n",
       "      <td>68</td>\n",
       "      <td>49</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26861</th>\n",
       "      <td>__no_feature</td>\n",
       "      <td>6557435</td>\n",
       "      <td>5231787</td>\n",
       "      <td>5195437</td>\n",
       "      <td>5114512</td>\n",
       "      <td>6921560</td>\n",
       "      <td>5789783</td>\n",
       "      <td>6671101</td>\n",
       "      <td>5483469</td>\n",
       "      <td>7671085</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26862</th>\n",
       "      <td>__ambiguous</td>\n",
       "      <td>1564</td>\n",
       "      <td>139460</td>\n",
       "      <td>139337</td>\n",
       "      <td>131957</td>\n",
       "      <td>437417</td>\n",
       "      <td>373195</td>\n",
       "      <td>458426</td>\n",
       "      <td>281404</td>\n",
       "      <td>499253</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26863</th>\n",
       "      <td>__too_low_aQual</td>\n",
       "      <td>397138</td>\n",
       "      <td>2113523</td>\n",
       "      <td>1645713</td>\n",
       "      <td>2013244</td>\n",
       "      <td>4980010</td>\n",
       "      <td>5489078</td>\n",
       "      <td>5347630</td>\n",
       "      <td>3620553</td>\n",
       "      <td>6400390</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26864</th>\n",
       "      <td>__not_aligned</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26865</th>\n",
       "      <td>__alignment_not_unique</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>26866 rows × 72 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                  GENE_SYMBOL  ENCODE_Os_1  ENCODE_Os_2  ENCODE_Os_3  \\\n",
       "0                        A1BG            0          102           91   \n",
       "1                    A1BG-AS1            1           41           28   \n",
       "2                        A1CF           23            0            0   \n",
       "3                         A2M            6           18           12   \n",
       "4                     A2M-AS1            4           18           26   \n",
       "...                       ...          ...          ...          ...   \n",
       "26861            __no_feature      6557435      5231787      5195437   \n",
       "26862             __ambiguous         1564       139460       139337   \n",
       "26863         __too_low_aQual       397138      2113523      1645713   \n",
       "26864           __not_aligned            0            0            0   \n",
       "26865  __alignment_not_unique            0            0            0   \n",
       "\n",
       "       ENCODE_Os_4  fibula_1  fibula_2  fibula_3  fibula_4  fibula_5  ...  \\\n",
       "0               90        30        21        41        17        80  ...   \n",
       "1               50        13         3        17         3        14  ...   \n",
       "2                0         5         3         1         3         5  ...   \n",
       "3               14      7667      7103      5452      2909     16884  ...   \n",
       "4               17        39        43        23        16        66  ...   \n",
       "...            ...       ...       ...       ...       ...       ...  ...   \n",
       "26861      5114512   6921560   5789783   6671101   5483469   7671085  ...   \n",
       "26862       131957    437417    373195    458426    281404    499253  ...   \n",
       "26863      2013244   4980010   5489078   5347630   3620553   6400390  ...   \n",
       "26864            0         0         0         0         0         0  ...   \n",
       "26865            0         0         0         0         0         0  ...   \n",
       "\n",
       "       230207-NORMAL  230207-OLF  230207-SPINOUS  230222-LF  230222-NORMAL  \\\n",
       "0                298         812             388        173            393   \n",
       "1                  0          11               0         60             54   \n",
       "2                 16          27               0          0              0   \n",
       "3              15229       31998           20449      77759          14516   \n",
       "4                  0           0               0         82             10   \n",
       "...              ...         ...             ...        ...            ...   \n",
       "26861              0           0               0          0              0   \n",
       "26862              0           0               0          0              0   \n",
       "26863              0           0               0          0              0   \n",
       "26864              0           0               0          0              0   \n",
       "26865              0           0               0          0              0   \n",
       "\n",
       "       230222-OLF  230222-SPINOUS  230316-DISH  230316-NORMAL  230316-OPLL  \n",
       "0             771             105         1151            642          480  \n",
       "1               6               0            1             55           12  \n",
       "2               4               0            3             77            2  \n",
       "3           21324            6991        36400          52678        66246  \n",
       "4              11               0           26             68           49  \n",
       "...           ...             ...          ...            ...          ...  \n",
       "26861           0               0            0              0            0  \n",
       "26862           0               0            0              0            0  \n",
       "26863           0               0            0              0            0  \n",
       "26864           0               0            0              0            0  \n",
       "26865           0               0            0              0            0  \n",
       "\n",
       "[26866 rows x 72 columns]"
      ]
     },
     "execution_count": 37,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_concat"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CSM_1</th>\n",
       "      <th>CSM_2</th>\n",
       "      <th>OPLL_1</th>\n",
       "      <th>OPLL_2</th>\n",
       "      <th>BM_1</th>\n",
       "      <th>BM_2</th>\n",
       "      <th>BM_3</th>\n",
       "      <th>NP_con_1</th>\n",
       "      <th>NP_con_2</th>\n",
       "      <th>NP_con_3</th>\n",
       "      <th>...</th>\n",
       "      <th>230207-NORMAL</th>\n",
       "      <th>230207-OLF</th>\n",
       "      <th>230207-SPINOUS</th>\n",
       "      <th>230222-LF</th>\n",
       "      <th>230222-NORMAL</th>\n",
       "      <th>230222-OLF</th>\n",
       "      <th>230222-SPINOUS</th>\n",
       "      <th>230316-DISH</th>\n",
       "      <th>230316-NORMAL</th>\n",
       "      <th>230316-OPLL</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>26461572</td>\n",
       "      <td>35110775</td>\n",
       "      <td>35277362</td>\n",
       "      <td>38755909</td>\n",
       "      <td>94826138</td>\n",
       "      <td>99398957</td>\n",
       "      <td>97591620</td>\n",
       "      <td>51135360</td>\n",
       "      <td>79018430</td>\n",
       "      <td>61898471</td>\n",
       "      <td>...</td>\n",
       "      <td>177248543</td>\n",
       "      <td>198142318</td>\n",
       "      <td>139643424</td>\n",
       "      <td>264841262</td>\n",
       "      <td>107705516</td>\n",
       "      <td>235989838</td>\n",
       "      <td>90419268</td>\n",
       "      <td>212121495</td>\n",
       "      <td>202178723</td>\n",
       "      <td>250859904</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1 rows × 69 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      CSM_1     CSM_2    OPLL_1    OPLL_2      BM_1      BM_2      BM_3  \\\n",
       "0  26461572  35110775  35277362  38755909  94826138  99398957  97591620   \n",
       "\n",
       "   NP_con_1  NP_con_2  NP_con_3  ...  230207-NORMAL  230207-OLF  \\\n",
       "0  51135360  79018430  61898471  ...      177248543   198142318   \n",
       "\n",
       "   230207-SPINOUS  230222-LF  230222-NORMAL  230222-OLF  230222-SPINOUS  \\\n",
       "0       139643424  264841262      107705516   235989838        90419268   \n",
       "\n",
       "   230316-DISH  230316-NORMAL  230316-OPLL  \n",
       "0    212121495      202178723    250859904  \n",
       "\n",
       "[1 rows x 69 columns]"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# 전체 raw read count의 총합 (data 총량)\n",
    "t = pd.DataFrame ( np.sum(df_concat.iloc[ : ,1 : ], axis = 0)  )\n",
    "t.astype (\"int\").T"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.12"
  },
  "orig_nbformat": 4
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
